clang/test/SemaCXX/builtin-assume-aligned-tmpl.cpp

```diff
@@ -52,6 +52,9 @@ T *atest3() __attribute__((assume_aligned(31, o))); // expected-error {{requeste
 template <typename T, int o>
 T *atest4() __attribute__((assume_aligned(32, o)));
 
+template<typename T>
+T atest5(int) __attribute__((assume_aligned(2)));
```

Comment:
This change causes a number of attributes to change their behavior (see all uses of `isValidPointerAttrType`), so we should test each of those.

Additionally, we should separately check (via a different declaration) that the diagnostic fires when instantiated.